include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -118,18 +43,25 @@ namespace alpaka
             //-----------------------------------------------------------------------------
             static auto declareVar(BlockSharedMemStMember<TDataAlignBytes> const& smem) -> T&
             {
-                smem.template alloc<T>();
-                return smem.template getLatestVar<T>();
+                auto* data = smem.template getVarPtr<T>(TuniqueId);
+
+                if(!data)
+                {
+                    smem.template alloc<T>(TuniqueId);
+                    data = smem.template getLatestVarPtr<T>();
+                }
+                ALPAKA_ASSERT(data != nullptr);
+                return std::ref(*data);
```

Comment:
Why are you constructing a `std::reference_wrapper<T>` here, that immediately converts to a `T&` again? Why not just `return *data`?

Reply:
I think here, as we indeed immediately convert to `T&`, it's indeed better to use just `return *data`